M365DSC Dev branch via Azure DevOps Pipeline fails to load - Microsoft.Graph.Authentication

<!--
    Your feedback and support is greatly appreciated, thanks for contributing!

    ISSUE TITLE:
    Please prefix the issue title with the resource name, e.g.
    'ResourceName: Short description of my issue'

    ISSUE DESCRIPTION (this template):
    Please provide information regarding your issue under each header below.
    Write N/A under any headers that do not apply to your issue, or if the
    information is not available.

    NOTE! Sensitive information should be obfuscated.

    PLEASE KEEP THE HEADERS.

    You may remove this comment block, and the other comment blocks,
    but please keep the headers.
-->
#### Details of the scenario you tried and the problem that is occurring
When using Azure DevOps pipeline with the following config, the output fails. A previous bug #668 fixed something, but now there is a new error message. 


#### Verbose logs showing the problem
```
2020-07-22T00:33:37.2928280Z VERBOSE: Loading module from path 'C:\Program 
2020-07-22T00:33:37.2929153Z Files\WindowsPowerShell\Modules\Microsoft.Graph.Authentication\0.7.1\Microsoft.Graph.Authentication.psd1'.
2020-07-22T00:33:37.3006092Z VERBOSE: Cannot verify the Microsoft .NET Framework version 4.7.2 because it is not included in the list of permitted 
2020-07-22T00:33:37.3038972Z versions.
2020-07-22T00:33:37.3102829Z VERBOSE: Loading 'FormatsToProcess' from path 'C:\Program 
2020-07-22T00:33:37.3105003Z Files\WindowsPowerShell\Modules\Microsoft.Graph.Authentication\0.7.1\Microsoft.Graph.Authentication.format.ps1xml'.
2020-07-22T00:33:37.4696651Z VERBOSE: Populating RepositorySourceLocation property for module Microsoft.Graph.Authentication.
2020-07-22T00:33:37.4745629Z VERBOSE: Loading module from path 'C:\Program 
2020-07-22T00:33:37.4747641Z Files\WindowsPowerShell\Modules\Microsoft.Graph.Authentication\0.7.1\./Microsoft.Graph.Authentication.psm1'.
2020-07-22T00:33:37.7792773Z Import-Module : An instance of GraphSession already exists. Call Initialize(Func<GraphSession>, bool) to overwrite it.
2020-07-22T00:33:37.7794012Z At C:\Program 
2020-07-22T00:33:37.7794613Z Files\WindowsPowerShell\Modules\Microsoft.Graph.Authentication\0.7.1\Microsoft.Graph.Authentication.psm1:13 char:9
2020-07-22T00:33:37.7795204Z + $null = Import-Module -Name (Join-Path $PSScriptRoot '.\bin\Microsoft ...
2020-07-22T00:33:37.7795667Z +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-07-22T00:33:37.7796143Z     + CategoryInfo          : NotSpecified: (:) [Import-Module], InvalidOperationException
2020-07-22T00:33:37.7796739Z     + FullyQualifiedErrorId : System.InvalidOperationException,Microsoft.PowerShell.Commands.ImportModuleCommand
2020-07-22T00:33:37.7797131Z  
2020-07-22T00:33:37.9304532Z ##[error]PowerShell exited with code '1'.
```


#### Suggested solution to the issue

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# Configure the Agent to run DSC configurations
winrm quickconfig -force 

# Install the M365DSC module
write-host "Installing M365DSC" 
Install-Module -Name Microsoft365DSC -Force  -verbose

#Install the Dev M365 DSC module
write-host "Installing Dev Branch"
Install-M365DSCDevBranch -verbose

```

#### The operating system the target node is running
<!--
    Please provide as much as possible about the target node, for example
    edition, version, build and language.
    On OS with WMF 5.1 the following command can help get this information.

    Get-ComputerInfo -Property @(
        'OsName',
        'OsOperatingSystemSKU',
        'OSArchitecture',
        'WindowsVersion',
        'WindowsBuildLabEx',
        'OsLanguage',
        'OsMuiLanguages')
-->
Microsoft hosted agent "windows-2019"


#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,10 @@
 # Change log for Microsoft365DSC
 
+## 1.20.1021.1
+
+* Fixed a permissions issue with the 
+  Install-M365DSCDevBranch cmdlet (Issue #699 & #826)
+
 ## 1.20.1014.1
 
 * TeamsVoiceRoute

--- a/Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1
+++ b/Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1
@@ -1385,9 +1385,11 @@ function Install-M365DSCDevBranch
     [CmdletBinding()]
     param()
     #region Download and Extract Dev branch's ZIP
+    Write-Host "Downloading the Zip package..." -NoNewline
     $url = "https://github.com/microsoft/Microsoft365DSC/archive/Dev.zip"
     $output = "$($env:Temp)\dev.zip"
     $extractPath = $env:Temp + "\O365Dev"
+    Write-Host "Done" -ForegroundColor Green
 
     Invoke-WebRequest -Uri $url -OutFile $output
 
@@ -1399,19 +1401,41 @@ function Install-M365DSCDevBranch
     $dependencies = $manifest.RequiredModules
     foreach ($dependency in $dependencies)
     {
-        Install-Module $dependency.ModuleName -RequiredVersion $dependency.RequiredVersion -Force -AllowClobber
-        Import-Module $dependency.ModuleName -Force
+        Write-Host "Installing {$($dependency.ModuleName)}..." -NoNewLine
+        $existingModule = Get-Module $dependency.ModuleName -ListAvailable | Where-Object -FilterScript {$_.Version -eq $dependency.RequiredVersion}
+        if ($null -eq $existingModule)
+        {
+            Install-Module $dependency.ModuleName -RequiredVersion $dependency.RequiredVersion -Force -AllowClobber | Out-Null
+        }
+        Import-Module $dependency.ModuleName -Force | Out-Null
+        Write-Host "Done" -ForegroundColor Green
     }
     #endregion
 
     #region Install M365DSC
+    Write-Host "Updating the Core Microsoft365DSC module..." -NoNewline
     $defaultPath = 'C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\'
     $currentVersionPath = $defaultPath + ([Version]$($manifest.ModuleVersion)).ToString()
+    
+    Copy-Item "$extractPath\Microsoft365DSC-Dev\Modules\Microsoft365DSC\*" `
+        -Destination $defaultPath -Recurse -Force
+
+    Import-Module ($defaultPath + "Microsoft365DSC.psd1") -Force | Out-Null
+    $oldModule = Get-Module 'Microsoft365DSC' | Where-Object -FilterScript {$_.ModuleBase -eq $currentVersionPath}
+    Remove-Module $oldModule -Force | Out-Null
     if (Test-Path $currentVersionPath)
     {
-        Remove-Item $currentVersionPath -Recurse -Confirm:$false
+        try
+        {
+            Remove-Item $currentVersionPath -Recurse -Confirm:$false -Force `
+                -ErrorAction Stop
+        }
+        catch
+        {
+            Write-Verbose $_
+        }
     }
-    Copy-Item "$extractPath\Microsoft365DSC-Dev\Modules\Microsoft365DSC" -Destination $currentVersionPath -Recurse -Force
+    Write-Host "Done" -ForegroundColor Green
     #endregion
 }
 

